google-cloud-spanner/src/main/java/com/google/cloud/spanner/connection/ReadWriteTransaction.java

```diff
@@ -331,7 +383,10 @@ void checkAborted() {
 
   void checkRolledBackToSavepoint() {
     if (this.rolledBackToSavepointException != null) {
-      if (savepointSupport == SavepointSupport.FAIL_AFTER_ROLLBACK) {
+      if (savepointSupport == SavepointSupport.FAIL_AFTER_ROLLBACK
+          && !((RollbackToSavepointException) this.rolledBackToSavepointException.getCause())
+              .getSavepoint()
+              .isAutoSavepoint()) {
```

Comment:
Would this throw a NPE when getSavepoint() returns null? I believe it is only set for emulator cases.

Reply:
It would if it was null, but it is set for all cases (so not only for the emulator case). I've added a `Preconditions.checkNotNull(savepoint)` check to the constructor of `RollbackToSavepointException` to guarantee it there as well.